src/main/java/cheryl/task/Deadline.java

```diff
@@ -0,0 +1,109 @@
+     */
+    public Deadline(String userInput) {
+        super(userInput.split("/by ")[0]);
+        this.date = new TimeProcessor(userInput.split("/by ")[1]).getDateTime();
+
+    }
+
+    /**
+     * Constructs a new Deadline task as a copy of an existing task, with a specified due date.
+     *
+     * @param other The task to copy.
+     * @param date The due date of the deadline task, as a string.
+     */
+    Deadline(Task other, String date) {
+        super(other);
+        this.date = new TimeProcessor(date).getDateTime();
+    }
+
+    /**
+     * Constructs a new Deadline task as a copy of an existing task, with a specified due date.
+     *
+     * @param other The task to copy.
+     * @param date The due date of the deadline task, as a {@link LocalDateTime} object.
+     */
+    public Deadline(Task other, LocalDateTime date) {
+        super(other);
+        this.date = date;
+    }
+
+    /**
+     * Constructs a new Deadline task with a specified completion status, description, and due date.
+     *
+     * @param check The completion status of the task ("[ ]" or "[X]").
+     * @param description The description of the deadline task.
+     * @param date The due date of the deadline task, as a string.
+     */
+    Deadline(String check, String description, String date) {
+        super(check, description);
+        this.date = new TimeProcessor(date).getDateTime();
+    }
+
+    /**
+     * Returns a string representation of the deadline task, including its type, status, description, and due date.
+     *
+     * @return A string in the format "[D][X] Task description (by: dueDate)".
+     */
+    public String toString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM/dd/yyyy HHmm");
+        return "[D]" + super.toString() + "(by: " + date.format(formatter) + ")";
+    }
+
+    /**
+     * Marks the deadline task as completed and returns a new Deadline instance with the updated status.
+     *
+     * @return A new deadline task marked as completed.
+     */
+    @Override
+    public Task mark() {
+        return new Deadline(super.mark(), this.date);
+    }
+
+    /**
+     * Unmarks the deadline task as incomplete and returns a new Deadline instance with the updated status.
+     *
+     * @return A new deadline task marked as incomplete.
+     */
+    @Override
+    public Task unmark() {
+        return new Deadline(super.unmark(), this.date);
+    }
+
+    /**
+     * Serializes the deadline task into a string format for storage or transmission.
+     * The format is "DEADLINE||[status]||[description]||[dueDate]".
+     *
+     * @return A serialized string representing the deadline task.
+     */
+    @Override
+    public String serialize() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+        return TaskType.DEADLINE + "||" + this.check + "||" + this.description + "||" + this.date.format(formatter);
```

Comment:
For a line of code that is long, perhaps it would be more readable to split that single line into a new line, then indent the new line by 8 spaces as per the coding standard recommendation.
```suggestion
        return TaskType.DEADLINE + "||" + this.check + "||" 
            + this.description + "||" + this.date.format(formatter);
```